glotaran/analysis/residual_calculation.py

```diff
@@ -37,7 +37,7 @@ def create_index_independent_ungrouped_residual(
             if callable(scheme.model.has_additional_penalty_function):
```

Comment:
We still need to bring this penalty function calculation at the same level of the preceeding for loop, after all the clps are calculated.